zebra-consensus/src/block/check.rs

```diff
@@ -148,77 +148,58 @@ pub fn subsidy_is_valid(block: &Block, network: &Network) -> Result<(), BlockErr
     let coinbase = block.transactions.first().ok_or(SubsidyError::NoCoinbase)?;
 
     // Validate funding streams
-    let Some(halving_div) = subsidy::general::halving_divisor(height, network) else {
-        // Far future halving, with no founders reward or funding streams
+
+    // Future halving, with no founders reward or funding streams
+    if subsidy::general::num_halvings(height, network) > 2 {
         return Ok(());
     };
 
     let canopy_activation_height = NetworkUpgrade::Canopy
         .activation_height(network)
         .expect("Canopy activation height is known");
 
-    // TODO: Add this as a field on `testnet::Parameters` instead of checking `disable_pow()`, this is 0 for Regtest in zcashd,
-    //       see <https://github.com/zcash/zcash/blob/master/src/chainparams.cpp#L640>
-    let slow_start_interval = if network.disable_pow() {
-        Height(0)
-    } else {
-        network.slow_start_interval()
-    };
-
-    if height < slow_start_interval {
-        unreachable!(
-            "unsupported block height: callers should handle blocks below {:?}",
-            slow_start_interval
-        )
-    } else if halving_div.count_ones() != 1 {
```

Comment:
`height < slow_start_interval` is checked in `num_halvings()` (and was checked in `halving_divisor()` before this change)

`halving_div.count_ones() != 1` where `halving_div` is `1 << num_halvings` should always be false if `num_halvings <= 2` (and should have always been false before this change since it would return early `if 1u64.checked_shl(num_halvings) == None`).